include/swift/Runtime/Metadata.h

```diff
@@ -1056,6 +1066,30 @@ struct GenericParameterDescriptorHeader {
   /// Flags for this generic parameter descriptor.
   GenericParameterDescriptorFlags Flags;
 
+  /// This is factored in a silly way because remote mirrors cannot directly
+  /// dereference the SuperClass field of class metadata.
+  uint32_t getOffset(const ClassMetadata *classMetadata,
+                     const ClassMetadata *superMetadata) const {
+    if (Flags.hasResilientSuperclass())
+      return superMetadata->getSizeInWords() + Offset;
+
+    return Offset;
+  }
+
+  uint32_t getOffset() const {
+    assert(!Flags.hasResilientSuperclass());
+    return Offset;
+  }
+
+  uint32_t getOffset(const TargetMetadata<Runtime> *metadata) const {
+    if (auto *classMetadata = llvm::dyn_cast<ClassMetadata>(metadata))
+      if (auto *superclass = classMetadata->SuperClass)
+        if (auto *superMetadata = llvm::dyn_cast<ClassMetadata>(superclass))
```

Comment:
Thanks, I've simplified it. I don't think an Objective-C class can be "resilient" in this manner because Objective-C class metadata always has a fixed size, no? There are no "members" in the sense of generic arguments, field offsets, vtable entries, etc.

Reply:
Right, I think that's reasonable.  Since that's true, you don't need to check for isTypeMetadata() in the case that you actually have to pull out the superclass size.